CONTRIBUTING.md

```diff
@@ -189,18 +189,24 @@ Testing the code using docker consists of executing the same file 3 times (you m
 First run `./scripts/container.sh --clear-cache`. Then run `./scripts/container.sh --build`. This starts a local docker image called `arviz`. Finally run the tests with `./scripts/container.sh --test`. This should be quite close to how the tests run on TravisCI.
 
 ### Using the Docker image interactively
-Once the Docker image is built with `./scripts/container.sh --build`, interactive containers can also be run. 
+Once the Docker image is built with `./scripts/container.sh --build`, interactive containers can also be run. Therefore, code can be edited and executed inside the docker container. Once finished, the results can be copied back to the host machine.
 
 To start a bash shell inside Docker, run:
 
     $ docker run -it arviz bash
 
-To start a jupyter notebook, there are two steps, first run:
+Alternatively, to start a jupyter notebook, there are two steps, first run:
 
     $ docker run --name jupyter-dock -it -d -p 8888:8888 arviz
     $ docker exec -it jupyter-dock pip install jupyter
     $ docker exec -it jupyter-dock jupyter notebook --ip 0.0.0.0 --no-browser --allow-root
 
 This will output something similar to `http://(<docker container id> or <ip>):8888/?token=<token id>`, and can be accessed at `http://localhost:8888/?token=<token id>`.
 
+Once the code has been edited in the docker container, it can be copied back to the host machine with the following command (on the host machine):
```

Comment:
This part I'm curious about. The container should share a filesystem with the host in the default configuration which means things shouldn't need to be copied. Is this not happening?

Reply:
Specifically the bind mount here should allow development in the container to be reflected on the host file system
https://github.com/arviz-devs/arviz/blob/master/scripts/container.sh#L26

I might have messed up though so if that's the case we should fix that rather than copy I think?

Reply:
Are you sure that the built containers will automatically mount the dir?  E.g., I would imagine you need to pass the `-v` in the above commands in this README?

Reply:
The bind in L26 is in the `--test` option. It does indeed bind the container with the working directory, but the bind has to be specified for every `docker run` instance. Between `--mount type=bind,...` and `-v`, the former is recommended even tough there are no plans to deprecate the latter.

I have updated the guide to use `--mount`.

Reply:
Now that I was thinking about it though, precisely in the `--test` flag case, wouldn't it be better to not bind the container?
 I am not totally sure about what exactly does the clear cache flag, but it looks like it basically removes caches from running the tests. If this is the case, removing the bind would solve these problems. In addition, as only the build flag is used in Travis, this won't have any effect on tests there nor on coveralls.

Reply:
Clearing the cache deletes pycache files and cached models. What was happening was that if a user ran a test on their local computer, python would create .pyc files and arviz would cache models. Subsequently if a user tried testing in docker, both python and/or pystan would throw an exception because the C code was now being run in an environment it wasn't compiled in.

By clearing the cache files, ArviZ and its dependencies in docker would have anything to load and this error would be passed.


As for whether to bind or not during tests, the trouble with not binding is that updates were made on the local filesystem, and somebody wanted to test those changes, they would have to rebuild the whole container just to reflect code changes, or do something like copy them in. I figured with ArviZ our recommendation should to bind containers to the local filesystem as our containers are meant for reproducible and rapid development, more so than anything else like app deployment or scaling.

This is all up for debate though! Let me know what you think

Reply:
It definitely is much easier with binding. I think that we should indicate that clear cache may have to be executed after testing though.

Reply:
Just to be explicit it might need to be run after the user tested on their local computer, but before they try testing in docker